dpctl/tests/test_tensor_asarray.py

```diff
@@ -623,3 +623,33 @@ def test_asarray_support_for_usm_ndarray_protocol(usm_type):
     assert x.dtype == y3.dtype
     assert y3.usm_data.reference_obj is None
     assert dpt.all(x[dpt.newaxis, :] == y3)
+
+
+@pytest.mark.parametrize("dt", [dpt.float16, dpt.float64, dpt.complex128])
+def test_asarray_to_device_with_unsupported_dtype(dt):
+    aspect = "fp16" if dt == dpt.float16 else "fp64"
+    try:
+        d0 = dpctl.select_device_with_aspects(aspect)
+    except dpctl.SyclDeviceCreationError:
+        pytest.skip("No device with aspect for test")
+    d1 = None
+    try:
```

Comment:
Would it be better to do through the loop?
```python
    for d in dpctl.get_devices():
        if d.default_selector_score < 0:
            pass
        try:
            d1 = dpctl.select_device_with_aspects(d.device_type.name, excluded_aspects=[aspect])
        except dpctl.SyclDeviceCreationError:
            pass
```

Reply:
sure, it's definitely less ugly